### DISCUSSION TITLE
(JS) How to set list value from file?

### DISCUSSION THREAD

**Discussion by piw-piw** [TurboWarp/packager#427] [Q&A] [answered]
How to set list value from file in project folder?
And vice versa?

**Comment by GarboMuffin** [comment, marked as answer]
most easily you just use right click -> import/export on a list watcher. If that doesn't work for you:

> How to set list value from file in project folder?

```js
fetch("file.txt")
  .then(r => r.text())
  .then(text => {
    vm.runtime.getTargetForStage().lookupVariableByNameAndType("list name", "list").value = text.split("\n");
  })
```

>  And vice versa?

You can get the list data from `vm.runtime.getTargetForStage().lookupVariableByNameAndType("list name", "list").value` and convert that to a string to download to your computer.
